Formula/k3d.rb

```diff
@@ -1,13 +1,14 @@
 class K3d < Formula
   desc "Little helper to run Rancher Lab's k3s in Docker"
   homepage "https://k3d.io"
-  url "https://github.com/rancher/k3d/archive/v5.0.3.tar.gz"
-  sha256 "85fff692fd3180fb9d050a0e25a99718cde7539e53ef0ffcb433fd19dd56061f"
+  url "https://github.com/rancher/k3d.git",
+    tag:      "v5.0.3",
+    revision: "89d979287e0cff56ac8e87e5e0ecab9dc3b6ba87"
   license "MIT"
 
   livecheck do
     url :stable
-    regex(/^v?(\d+(?:\.\d+)+)$/i)
+    strategy :github_latest
```

Comment:
Why is this needed?

Reply:
It made sense to me after checking the docs when I saw it in other formulae, as per https://rubydoc.brew.sh/Homebrew/Livecheck/Strategy/GithubLatest.html (and the regex is the default in that strategy).

Reply:
Does the current livecheck produce wrong results? If it doesn't, let's keep it as it is. `:github_latest` gets rate limited so we only use it when we actually have evidence that it's needed.

Reply:
Ah if it's rate-limited then I'll change it back.
As most other changes it was inspired by e.g. the k3sup and k9s formula.
But this makes sense 👍